### PR TITLE
(SERVER-2543) add plan api docs

### PR DESCRIPTION
**Pull request by steveax**
In a previous commit [1] the plan list and plan metadata endpoints
were added. This commit adds API docs for those new endpoints.

[1]: 531ce7f86aeae13317e11b7569996b74e0895620

### CODE REVIEW

File: documentation/puppet-api/v3/plan_detail.markdown [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+```
+GET /puppet/v3/plans/doesnotexist/planname?environment=env
+
+HTTP/1.1 404 Not Found
+
+Could not find module 'doesnotexist'
+```
+
+#### Task does not exist or does not have a valid name
```
Comment (mollykmcglone) [inline documentation/puppet-api/v3/plan_detail.markdown]: Task -> Plan

File: documentation/puppet-api/v3/plans.markdown [hunk 2]
```diff
@@ -0,0 +1,119 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Plans"
+canonical: "/puppetserver/latest/puppet-api/v3/plans.html"
+---
+
+### Uses `application/json` Content-Type
+
+The Content-Type in the response to an task API query is
```
Comment (mollykmcglone) [inline documentation/puppet-api/v3/plans.markdown]: task -> plan

File: documentation/puppet-api/v3/plans.markdown [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+### Supported Formats
+
+JSON
+
+### Query Parameters
+
+Provide one parameter to the GET request:
+
+* `environment`: Only the task information pertaining to the specified
```
Comment (mollykmcglone) [inline documentation/puppet-api/v3/plans.markdown]: task -> plan

File: documentation/puppet-api/v3/plan_detail.markdown (hunk 1 above)
Comment (steveax) [reply documentation/puppet-api/v3/plan_detail.markdown]: good catch!

**Comment by justinstoller** [comment]
Couldn't find the skeletor team alias, so I pinged individuals. Feel free to merge.

Sorry, I should have called out where our docs live when the original PR went in!

File: documentation/puppet-api/v3/plan_detail.markdown [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+---
+
+### Uses `application/json` Content-Type
+
+The Content-Type in the response to an plan API query is
+`application/json`.
+
+## `GET /puppet/v3/plans/:module/:plan?environment=:environment`
+
+(Introduced in Puppet Server 5.1.0.)
```
Comment (branan) [inline documentation/puppet-api/v3/plan_detail.markdown]: I think puppetserver got tagged this week for the platform release, so we can find the right 6.Y?

Comment (steveax) [reply documentation/puppet-api/v3/plan_detail.markdown]: Good catch, looks like 6.6.0, updated!

**Comment by steveax** [comment]
OK, I think this is GtG now.

**Comment by McdonaldSeanp** [comment]
I'll merge this once travis passes

**Pull request merged**